File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/HowTo/GetDocumentId.cs
using Raven.Client.Document;

namespace Raven.Documentation.CodeSamples.ClientApi.Session.HowTo
{
	public class GetDocumentId
	{
		#region get_document_id_3
		public class Comment
		{
			public string Author { get; set; }

			public string Message { get; set; }
		}
		#endregion

		private interface IFoo
		{
			#region get_document_id_1
			string GetDocumentId(object entity);
			#endregion
		}

		public GetDocumentId()
		{
			using (var store = new DocumentStore())
			{
				Comment comment = null;

				using (var session = store.OpenSession())
				{
					#region get_document_id_2
					var commentId = session
						.Advanced
						.GetDocumentId(comment); // e.g. comments/1
					#endregion
				}
			}
		}
	}
}

File: Documentation/3.5/Samples/java/src/test/java/net/ravendb/clientapi/session/howto/GetDocumentId.java
package net.ravendb.clientapi.session.howto;

import net.ravendb.client.IDocumentSession;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;


public class GetDocumentId {

  //region get_document_id_3
  public static class Comment {
    private String author;
    private String message;
    public String getAuthor() {
      return author;
    }
    public void setAuthor(String author) {
      this.author = author;
    }
    public String getMessage() {
      return message;
    }
    public void setMessage(String message) {
      this.message = message;
    }
  }
  //endregion

  @SuppressWarnings("unused")
  private interface IFoo {
    //region get_document_id_1
    public String getDocumentUrl(Object entity);
    //endregion
  }

  public GetDocumentId() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      Comment comment = null;
      try (IDocumentSession session = store.openSession()) {
        //region get_document_id_2
        session.advanced().getDocumentId(comment); // e.g. comments/1
        //endregion
      }
    }
  }
}
